### PR TITLE
allow disabling config syncer rbac and sa 

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

Allow disabling configsyncer rbac and service account with a much granular configuration

## Related Issues

- https://github.com/astronomer/issues/issues/6728

## Testing

QA can independently control the creation of configsyncer rbac, service account when global.rbacEnabled is set to true

## Merging

cherry-pick to release-0.36

### CODE REVIEW

**Comment by pgvishnuram** [comment]
@rishkarajgi yes tests updated

**Review by rishkarajgi — APPROVED** [review]

File: charts/astronomer/templates/config-syncer/config-syncer-role.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@
 # 2. Create a Cluster Role if namespacePools is disabled
 # 3. Create roles for each namespaces in the namespacePool (+ astronomer namespace) if enabled.
 {{- if .Values.configSyncer.enabled }}
+{{- if and .Values.global.rbacEnabled .Values.configSyncer.serviceAccount.create }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/config-syncer/config-syncer-role.yaml:9]: These can be reduced into one line.

File: charts/astronomer/templates/config-syncer/config-syncer-rolebinding.yaml [hunk 2]
```diff
@@ -6,6 +6,7 @@
 # 2. Create ClusterRoleBinding if namespacePools disabled
 # 3. Create RoleBinding for each namespace in the namespacePool (+ astronomer namespace) if enabled
 {{- if .Values.configSyncer.enabled }}
+{{- if and .Values.global.rbacEnabled .Values.configSyncer.serviceAccount.create }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/config-syncer/config-syncer-rolebinding.yaml:9]: These can be reduced into one line.

File: charts/astronomer/templates/config-syncer/config-syncer-serviceaccount.yaml [hunk 3]
```diff
@@ -1,7 +1,8 @@
 ###############################
 ## Astronomer ServiceAccount ##
 ###############################
-{{- if and .Values.global.rbacEnabled .Values.configSyncer.enabled }}
+{{- if .Values.configSyncer.enabled  }}
+{{- if and .Values.global.rbacEnabled .Values.configSyncer.serviceAccount.create }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/config-syncer/config-syncer-serviceaccount.yaml:5]: These can be reduced into one line.

File: tests/chart_tests/test_astronomer_config_syncer.py [hunk 4]
```diff
@@ -278,3 +278,23 @@ def test_astronomer_config_syncer_disabled(self, kube_version):
         )
 
         assert len(docs) == 0
+
+    def test_astronomer_config_syncer_and_global_rbac_enabled_with_sa_create_disabled(self, kube_version):
+        """Test that config syncer service account, rbac  is disabled when config syncer and global rbac is enabled."""
+
+        # First rbacEnabled set to true and namespacePools disabled, should create a ClusterRole and ClusterRoleBinding
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_config_syncer.py:285]: This comment implies that there will be a "Second". Can we improve or remove this comment? Do we need to add another test?

File: tests/chart_tests/test_astronomer_config_syncer.py [hunk 5]
```diff
@@ -278,3 +278,23 @@ def test_astronomer_config_syncer_disabled(self, kube_version):
         )
 
         assert len(docs) == 0
+
+    def test_astronomer_config_syncer_and_global_rbac_enabled_with_sa_create_disabled(self, kube_version):
+        """Test that config syncer service account, rbac  is disabled when config syncer and global rbac is enabled."""
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_config_syncer.py:283]: This wording seems better, assuming I understood the original comment correctly.
```suggestion
        """Test that config syncer service account and rbac are disabled when config syncer and global rbac is enabled."""
```

File: charts/astronomer/templates/config-syncer/config-syncer-role.yaml (hunk 1 above)
Comment (danielhoherd) [inline charts/astronomer/templates/config-syncer/config-syncer-role.yaml:9]:
```suggestion
{{- if and .Values.configSyncer.enabled .Values.global.rbacEnabled .Values.configSyncer.serviceAccount.create }}
```

File: charts/astronomer/templates/config-syncer/config-syncer-role.yaml [hunk 6]
```diff
@@ -45,3 +46,4 @@ rules:
 {{ end }}
 {{- end }}
 {{- end }}
+{{- end }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/config-syncer/config-syncer-role.yaml:49]:
```suggestion
{{- end }}
```

File: charts/astronomer/templates/config-syncer/config-syncer-rolebinding.yaml (hunk 2 above)
Comment (danielhoherd) [inline charts/astronomer/templates/config-syncer/config-syncer-rolebinding.yaml:9]:
```suggestion
{{- if and .Values.configSyncer.enabled .Values.global.rbacEnabled .Values.configSyncer.serviceAccount.create }}
```

File: charts/astronomer/templates/config-syncer/config-syncer-rolebinding.yaml [hunk 7]
```diff
@@ -42,3 +43,4 @@ subjects:
 {{ end }}
 {{- end }}
 {{- end }}
+{{- end }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/config-syncer/config-syncer-rolebinding.yaml:46]:
```suggestion
```

File: charts/astronomer/templates/config-syncer/config-syncer-serviceaccount.yaml (hunk 3 above)
Comment (danielhoherd) [reply charts/astronomer/templates/config-syncer/config-syncer-serviceaccount.yaml:5]:
```suggestion
{{- if and .Values.configSyncer.enabled .Values.global.rbacEnabled .Values.configSyncer.serviceAccount.create }}
```

File: charts/astronomer/templates/config-syncer/config-syncer-serviceaccount.yaml [hunk 8]
```diff
@@ -19,3 +20,4 @@ metadata:
     argocd.argoproj.io/sync-wave: "-1"
     {{- end }}
 {{- end }}
+{{- end }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/config-syncer/config-syncer-serviceaccount.yaml:23]:
```suggestion
```

File: tests/chart_tests/test_astronomer_config_syncer.py (hunk 4 above)
Comment (danielhoherd) [reply tests/chart_tests/test_astronomer_config_syncer.py:285]: I unresolved this comment because I have not seen the answer to my questions and concerns.

**Pull request merged**